### ISSUE TITLE
Requests and questions

### ISSUE THREAD

**Issue by T-wow** [Hope1337/YOWOv3#14]
Hello author, may I ask if it is possible to add features for recovery training and DDP training. May I ask if training from scratch can achieve your level of accuracy

**Comment by T-wow** [comment]
Please add a code module that does not include pretrained models for both branches.Thanks

**Comment by Hope1337** [comment]
Hi @T-wow , 

- About add features for recovery training and DDP training : I'm not sure what the term "recovery training" means (even after searching, it doesn't seem to be common), but if you are referring to continuing training from a checkpoint, I had planned to do this before. The task involves an EMA epoch with a decay factor, just saving and loading would be sufficient. I will consider incorporating it into the project. Currently, I only have a single GPU, so I can't test DPP training, I'm sorry.

- About accuracy when training from scratch : Certainly, you can absolutely achieve equivalent mAP (or even higher) when training from scratch. However, it's important to note that due to the random factor, the results may vary (be worse, equal, or better), so you might need to try training a few times to assess the performance.

- About " add a code module that does not include pretrained" : I still don't understand your point when you mention "does not include pretrained," please explain it clearly.

Thank you for asking the question.

**Comment by T-wow** [comment]
> Hi @T-wow ,
> 
> * About add features for recovery training and DDP training : I'm not sure what the term "recovery training" means (even after searching, it doesn't seem to be common), but if you are referring to continuing training from a checkpoint, I had planned to do this before. The task involves an EMA epoch with a decay factor, just saving and loading would be sufficient. I will consider incorporating it into the project. Currently, I only have a single GPU, so I can't test DDP training, I'm sorry.
> * About accuracy when training from scratch : Certainly, you can absolutely achieve equivalent mAP (or even higher) when training from scratch. However, it's important to note that due to the random factor, the results may vary (be worse, equal, or better), so you might need to try training a few times to assess the performance.
> * About " add a code module that does not include pretrained" : I still don't understand your point when you mention "does not include pretrained," please explain it clearly.
> 
> Thank you for asking the question.
“a code module that does not include pretrained” refers to training the 2D and 3D branch models from scratch without loading pretrained weights.
At the same time, I would like to ask whether training both branches from scratch can achieve your expected results. Thank you for your response.

**Comment by T-wow** [comment]
@Hope1337  Thank you for your response.
“a code module that does not include pretrained" refers to training the 2D and 3D branch models from scratch without loading pretrained weights.
At the same time, I would like to ask whether training both branches from scratch can achieve your expected results.

**Comment by Hope1337** [comment]
@T-wow 
In my experiments, training two backbones from scratch always results in a significant drop in mAP. The reason for this is easily understandable as pretraining is done on a very large dataset and generalizes much better.

**Comment by T-wow** [comment]
@Hope1337 
Thank you for your response. I would like to make some improvements to the network structure based on your work. I hope you can add the functionality to train backbone2D and backbone3D from scratch. I greatly appreciate it.
I encountered an error after setting config['BACKBONE2D']['YOLOv8']['PRETRAIN'][ver] to None, so I am looking forward to the release of this feature.

**Comment by T-wow** [comment]
@Hope1337 Thank you for your response.

**Comment by PEACHTTT** [comment]
> @T-wow Adding this feature requires changing the structure of the config file and necessitates modifying the entire config file in the checkpoint to ensure consistency. However, this task is not difficult; I can provide you with a guide:
> 
>     * First, since the config is a YAML file, you need to use **null** instead of None in the config file. After being read from the YAML file, the variable will automatically be converted to None.
> 
>     * Each backbone has a function called `load_pretrain` to load the pretrained model, for example, in the file `model/backbone2D/YOLOv8.py` [here](https://github.com/Hope1337/YOWOv3/blob/480bf188b4bd6b605a8538cfbbbdaaa11fc3be2a/model/backbone2D/YOLOv8.py#L154) :
> 
> 
> ```python
>     def load_pretrain(self):
>         state_dict = self.state_dict()
> 
>         pretrain_state_dict = torch.load(self.pretrain_path)
>         
>         for param_name, value in pretrain_state_dict.items():
>             if param_name not in state_dict:
>                 continue
>             state_dict[param_name] = value
>             
>         self.load_state_dict(state_dict)
> 
>         print("backbone2D : YOLOv8 pretrained loaded!", flush=True)
> ```
> 
> You just need to add a few lines of code at the beginning of the `load_pretrain` function, for example :
> 
> ```python
> if self.pretrain_path is None:
>    print("this backbone is training from scratch")
>    return
> ```
> 
> Similarly with other backbones.

When I changed the config file and backbone file codes  like shufflenetv2 as your guidance ,I set a print but it seems the 3D model is still the pretrained one  , I noticed that the build_backbone3D use code bellow to get model
![image](https://github.com/user-attachments/assets/a3eebd9b-ecfb-43f6-9a48-e769aafcb4b9)
~~~IN build_shufflenetv2,the pretrained path is still appointed directly, so I am confused about how to retrain the YOWOV3 totally
Thanks for your replying!
![image](https://github.com/user-attachments/assets/b596613f-e0ca-4e61-a26a-5e84b1408390)

**Comment by T-wow** [comment]
First, you need to change the pretrain_path in the configuration file to null and set the PRETRAIN for both backbones to null. Then, you can follow the method provided by the blogger to add the relevant code for not loading the pre-trained model in both backbones:
def load_pretrain(self):
    state_dict = self.state_dict()

    if self.pretrain_path is not None:
        pretrain_state_dict = torch.load(self.pretrain_path)
        
        for param_name, value in pretrain_state_dict.items():
            if param_name not in state_dict:
                continue
            state_dict[param_name] = value
        
        self.load_state_dict(state_dict)

        print("backbone2D : YOLOv8 pretrained loaded!", flush=True)
    else:
        print("No pretrain path provided, training from scratch.", flush=True)
        This way, YOWOv3 can be trained from scratch as a whole.

This way, YOWOv3 can be trained from scratch as a whole.

**Comment by PEACHTTT** [comment]
Thanks!
